core/src/main/scala/io/finch/internal/ToResponse.scala

```diff
@@ -46,11 +46,22 @@ object ToResponse extends LowPriorityToResponseInstances {
 
   implicit def asyncBufToResponse[CT <: String](implicit
     w: Witness.Aux[CT]
-  ): Aux[AsyncStream[Buf], CT] = instance { a =>
+  ): Aux[AsyncStream[Buf], CT] = asyncStreamResponseBuilder(identity)
+
+  implicit def asyncStreamToResponse[A, CT <: String](implicit
+    e: Encode.Aux[A, CT],
+    w: Witness.Aux[CT]
+  ): Aux[AsyncStream[A], CT] = asyncStreamResponseBuilder(data => e(data) concat Buf.Utf8("\n"))
+
+  private def asyncStreamResponseBuilder[A, CT <: String](writer: A => Buf)(implicit
```

Comment:
This would work for sure, but my take is we don't need `asyncStreamResponseBuilder` since there is already `Encode[Buf]` defined as `identity`. Just remove the previous version and keep your one - everything should work.